### ISSUE TITLE
Error in openapi schema for Enum

### ISSUE THREAD

**Issue by szagi3891** [poem-web/poem#121]
## Expected Behavior

A different type is expected for the enum:
```
"geoRuleType": {
  "type": "string",
  "enum": [
    "DENY",
    "ALLOW"
  ]
}
```

## Actual Behavior

```
"GeoRuleType": {
  "type": "GeoRuleType",
  "enum": [
    "DENY",
    "ALLOW"
  ]
},
```

## Steps to Reproduce the Problem

1. You have to clone https://github.com/szagi3891/poem-test
2. And then run with cargo run

## Specifications

- Version: poem = "1.0.38", poem-openapi = "1.0.41"
- Platform: MacOs
- Subsystem: 

```
{
  "openapi": "3.0.0",
  "info": {
    "title": "Oneof",
    "version": "1.0"
  },
  "servers": [],
  "tags": [],
  "paths": {
    "/api1/hello": {
      "post": {
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "type": "object",
                "properties": {
                  "type": {
                    "type": "string",
                    "enum": [
                      "A",
                      "B"
                    ]
                  }
                },
                "oneOf": [
                  {
                    "$ref": "#/components/schemas/A"
                  },
                  {
                    "$ref": "#/components/schemas/B"
                  }
                ],
                "discriminator": {
                  "propertyName": "type"
                }
              }
            }
          },
          "required": true
        },
        "responses": {
          "200": {
            "description": "Created successfully",
            "content": {
              "application/json": {
                "schema": {
                  "type": "integer",
                  "format": "uint64"
                }
              }
            }
          },
          "403": {
            "description": "Permission denied",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/Forb"
                }
              }
            }
          },
          "500": {
            "description": "Internal error"
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "GeoRuleType": {
        "type": "GeoRuleType",
        "enum": [
          "DENY",
          "ALLOW"
        ]
      },
      "B": {
        "type": "object",
        "required": [
          "v3",
          "list"
        ],
        "properties": {
          "v3": {
            "type": "number",
            "format": "float32"
          },
          "list": {
            "type": "object",
            "additionalProperties": {
              "type": "integer",
              "format": "uint64"
            }
          },
          "sport": {
            "$ref": "#/components/schemas/SportModel"
          }
        }
      },
      "Forb": {
        "type": "object",
        "required": [
          "message",
          "age"
        ],
        "properties": {
          "message": {
            "type": "string"
          },
          "age": {
            "type": "integer",
            "format": "int32"
          }
        }
      },
      "A": {
        "type": "object",
        "required": [
          "v1",
          "v2"
        ],
        "properties": {
          "v1": {
            "type": "integer",
            "format": "int32"
          },
          "v2": {
            "type": "string"
          }
        }
      },
      "SportModel": {
        "type": "object",
        "required": [
          "id",
          "name",
          "displayOrder",
          "geoRuleType"
        ],
        "properties": {
          "id": {
            "type": "string"
          },
          "name": {
            "type": "string"
          },
          "displayOrder": {
            "type": "integer",
            "format": "uint64"
          },
          "geoRuleType": {
            "$ref": "#/components/schemas/GeoRuleType"
          }
        }
      }
    }
  }
}
```

**Comment by szagi3891** [comment]
I have validated this specification on https://editor.swagger.io/
This page returns such an error message:

```
Structural error at components.schemas.GeoRuleType.type
should be equal to one of the allowed values
allowedValues: array, boolean, integer, number, object, string
```

**Comment by szagi3891** [comment]
```
pub enum State {
    #[serde(rename = "open")]
    Open
}
```

How to handle the renaming of a field in a poem ?

**Comment by sunli829** [comment]
Thank you very much, fixed in `1.0.43` 🙂

```rust
#[derive(Enum)]
enum MyEnum {
     CreateUser,
     #[oai(rename = "delete_user")]
     DeleteUser,
}
```

**Comment by szagi3891** [comment]
Works beautifully :)

And are there any equivalents for these attributes ?
#[serde(untagged)]
#[serde(deny_unknown_fields)]

**Comment by sunli829** [comment]
> #[serde(deny_unknown_fields)]

I want to deny unknown fields by default, what do you think? @szagi3891

**Comment by szagi3891** [comment]
Yes, it would be nice to have this option.

Would it be possible to make an equivalent for this switch ?
#[serde(untagged)]

**Comment by sunli829** [comment]
Sorry, can you tell me what `#[serde(untagged)]` does? 😁

**Comment by szagi3891** [comment]
https://serde.rs/enum-representations.html#untagged

Then you can have many different variants of objects with different types. Something like OneOf but without the field that defines their type.

**Comment by sunli829** [comment]
How to define this type in the spec?

**Comment by sunli829** [comment]
I guess you want this?

```rust
#[serde(tag = "type")]
enum Obj {
    A {
        a: i32,
    }
    B {
        b: i32
    }
}
```

**Comment by szagi3891** [comment]
This is a sample schema:

```
"sport": {
   "nullable":true,
   "oneOf":[
      {
         "type":"string"
      },
      {
         "items":{
            "type":"string"
         },
         "type":"array"
      },
      {
         "properties":{
            "ne":{
               "items":{
                  "type":"string"
               },
               "type":"array"
            }
         },
         "type":"object",
         "required":[
            "ne"
         ]
      }
   ]
},
```

In this case of use, the field that specifies the type of variant of the enum is not used at all

**Comment by szagi3891** [comment]
This schema above, is generated  from such a rustic structure|:

```
#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(untagged)]
pub enum EventsQuerySport {
    Simple(Arc<String>),
    Vec(Vec<Arc<String>>),
    Not {
        ne: Vec<Arc<String>>
    }
}
```

**Comment by sunli829** [comment]
> deny_unknown_fields

Released in `v1.0.50` 🙂
